addons/mrp/models/stock_move.py

```diff
@@ -140,6 +140,16 @@ class StockMove(models.Model):
     product_qty_available = fields.Float('Product On Hand Quantity', related='product_id.qty_available')
     product_virtual_available = fields.Float('Product Forecasted Quantity', related='product_id.virtual_available')
     description_bom_line = fields.Char('Kit', compute='_compute_description_bom_line')
+    manual_consumption = fields.Boolean('Manual Consumption',
+        compute='_compute_manual_consumption', store=True,
+        help="When activated, then the registration of consumption for that component is recorded manually exclusively.\n"
+             "If not activated, and any of the components consumption is edited manually on the manufacturing order, Odoo assumes manual consumption also.")
+
+    @api.depends('state')
+    def _compute_manual_consumption(self):
+        for move in self:
+            if move.state == 'draft':
+                move.manual_consumption = move.bom_line_id.manual_consumption
```

Comment:
Instead of the `_onchange_quantity_done`, can't you do all in the compute here ?
For example:
```suggestion
    @api.depends('quantity_done', 'state')
    def _compute_manual_consumption(self):
        for move in self:
            if not move.raw_material_production_id or move.manual_consumption:
                continue
            if move.state == 'draft':
                move.manual_consumption = move.bom_line_id.manual_consumption
            else:
                move.manual_consumption = move.should_consume_qty != move.quantity_done
```
Can you try and see if there is issues with this solution ?

Reply:
The issue is when `quantity_done` get updated, we will remove all sml then create new sml for it. And `quantity_done` is also computed, so when all sml are removed, its value will be 0 and make `move.should_consume_qty != move.quantity_done` always true. `onchange` doesn't have this issue, so arm suggested split this into two functions.